exchangelib/util.py

```diff
@@ -514,11 +514,11 @@ def post_ratelimited(protocol, session, url, headers, data, allow_redirects=Fals
             try:
                 data = data.encode('utf-8')
             except UnicodeDecodeError:
-                import chardet
-                encoding_info = chardet.detect(data)
-                if encoding_info['encoding'] in ['ISO-8859-1', 'Windows-1252']:
```

Comment:
So I had added this line because previously we were detecting encodings with chardet and always encoding with what it had detected before sending to the server. But if the encoding was detected to be 'ISO-8859-1' or 'Windows-1252' the server was throwing ErrorSchemaValidation errors (https://app.clubhouse.io/nylas/story/26621/create-event-schema-validation-error). Adding this line reduced this error significantly but not completely and we still have lingering UnicodeDecodeErrors. Because of how opaque these errors are I'm using my best guess that even if the encoding is detected to be something other than the two above we get UnicodeDecodeErrors down the line if we encode based on the detected encoding. So I figured we should try one more time to encode with utf-8 before relying on chardet.

Reply:
Sounds good! Ty for the explanation :)